analyses/tmb-compare/compare-tmb-calculations.Rmd

```diff
@@ -0,0 +1,217 @@
+  dplyr::rename(tmb_d3b = TMB, 
+                tmb_ccdl = tmb)
+```
+
+## Correlate the TMB calculations
+
+Correlate with Pearson's. 
+
+```{r}
+cor.test(all_tmb$tmb_d3b, all_tmb$tmb_ccdl, method = "pearson")
+```
+
+It's super skewed, so let's also try by Spearman's. 
+
+```{r}
+cor.test(all_tmb$tmb_d3b, all_tmb$tmb_ccdl, method = "spearman")
+```
+
+Let's also try this for `mutation_count`. 
+
+```{r}
+cor.test(all_tmb$mutation_count, all_tmb$count, method = "spearman")
+```
+
+Results are about the same as with TMB. 
+
+## Plot this but separate by experimental strategy
+
+Plot this separately by `experimental_strategy` (while noting that the WXS samples are also all of the same histology).
+
+```{r}
+ggplot2::ggplot(all_tmb, 
+                ggplot2::aes(x = tmb_ccdl, y = tmb_d3b, color = disease)) + 
+  ggplot2::geom_point() + 
+  ggplot2::facet_wrap(~experimental_strategy.x, scales = "free") + 
+  ggplot2::theme_classic() + 
+  ggplot2::theme(legend.position = "none")
+```
+
+Mostly I'm seeing the ranges for the TMBs are very different.
+CCDL TMB is a much larger range (0 - ~300), while D3B TMB is smaller range (0 - ~5). 
+But the outliers also make this hard to interpret. 
+Let's try this with counts to see if its the same. 
+
+```{r}
+ggplot2::ggplot(all_tmb, 
+                ggplot2::aes(x = mutation_count, y = count, color = disease)) + 
+  ggplot2::geom_point() + 
+  ggplot2::facet_wrap(~experimental_strategy.x, scales = "free") + 
+  ggplot2::theme_classic() + 
+  ggplot2::theme(legend.position = "none")
+```
+
+Both look the same.
+This is super skewed, still not giving great visuals; let's try something else. 
+
+## Calculate percent difference in TMB
+
+```{r}
+all_tmb <- all_tmb %>%
+  dplyr::mutate(
+    diff = abs(tmb_d3b - tmb_ccdl),
+    perc_diff  = diff/max(tmb_ccdl, tmb_d3b)
+    ) 
+```
+
+Plot the percent difference. 
+
+```{r}
+ggplot2::ggplot(all_tmb, 
+                ggplot2::aes(x = experimental_strategy.x, y = perc_diff, fill = experimental_strategy.x)) + 
+  ggplot2::geom_violin() + 
+  ggplot2::theme_classic() +
+  ggplot2::theme(legend.position = "none") 
+```
+
+### Which samples are really differing in TMB calculations 
+
+```{r}
+all_tmb %>% 
+  dplyr::filter(perc_diff > .05)
+```
+
+Most are HGAT. 
+
+## How different are the counts?
+
+```{r}
+all_tmb %>% 
+  dplyr::filter(perc_diff > .05) %>%
+  ggplot2::ggplot(ggplot2::aes(x = mutation_count, y = count, color = short_histology)) + 
+  ggplot2::geom_point() + 
+  ggplot2::ylab("D3B mutation count") +
+  ggplot2::xlab("CCDL mutation count") + 
+  ggplot2::theme_classic()
+```
+
+Counts look mostly the same. 
+
+## Calculate percent difference in BED size
+
+```{r}
+all_tmb <- all_tmb %>%
+  dplyr::mutate(
+    bed_diff = abs(region_size - bedlength),
+    bed_perc_diff  = bed_diff/max(region_size, bedlength)
+    ) 
+```
+
+Let's make a table of this.
+
+```{r}
+table(all_tmb$region_size, all_tmb$bedlength)
```

Comment:
I don't think a `table()` is what we want here. We really just want to see the parings of region size and bed length. Something like:
```
all_tmb %>%
  dplyr::select("region_size", "bed_length") %>%
  dplyr::distinct()
```

Or if you want counts:
```
all_tmb %>%
  dplyr::group_by("region_size", "bed_length") %>%
  dplyr::tally()
```